sylvia-derive/src/contract/communication/reply.rs

```diff
@@ -180,6 +253,33 @@ impl<'a> ReplyData<'a> {
             }
         }
     }
+
+    fn emit_cw_reply_on(&self) -> TokenStream {
+        let sylvia = crate_module();
+        let is_always = self
+            .handlers
+            .iter()
+            .any(|(_, reply_on)| reply_on == &ReplyOn::Always);
+        let is_success = self
+            .handlers
+            .iter()
+            .any(|(_, reply_on)| reply_on == &ReplyOn::Success);
+        let is_failure = self
+            .handlers
+            .iter()
+            .any(|(_, reply_on)| reply_on == &ReplyOn::Failure);
+
+        if is_always || (is_success && is_failure) {
+            quote! { #sylvia ::cw_std::ReplyOn::Always }
+        } else if is_success {
+            quote! { #sylvia ::cw_std::ReplyOn::Success }
+        } else if is_failure {
+            quote! { #sylvia ::cw_std::ReplyOn::Error }
+        } else {
+            // This should never happen
```

Comment:
It will earlier while parsing. We support `Success`, `Failure`, and `Always`, as specifying logic that would `Never` be called makes no sense.

I will add better comment there.